aequilibrae/utils/db_utils.py

```diff
@@ -32,9 +33,19 @@ def __init__(self, db: Union[str, Path, Connection], commit: bool = True, missin
 
             **missing_ok** (:obj:`bool`): Boolean indicating that the db is not expected to exist yet
         """
-        if isinstance(db, str) or isinstance(db, Path):
-            db = safe_connect(db, missing_ok)
-        self.conn = db
+        from aequilibrae.utils.spatialite_utils import connect_spatialite
+
+        def get_conn() -> sqlite3.Connection:
+            if spatial:
+                if not isinstance(db, (str, PathLike)):
+                    raise Exception("You must provide a database path to connect to spatialite")
+                return connect_spatialite(db, missing_ok)
+            elif isinstance(db, (str, PathLike)):
+                return safe_connect(db, missing_ok)
+            else:
+                return db
+
+        self.conn = get_conn()
```

Comment:
```suggestion
        if spatial:
            if not isinstance(db, (str, PathLike)):
                raise Exception("You must provide a database path to connect to spatialite")
            self.conn = connect_spatialite(db, missing_ok)
        elif isinstance(db, (str, PathLike)):
            self.conn = safe_connect(db, missing_ok)
        else:
            self.conn = db
```

This is more personal preference but I thought the `get_conn` method was used else where until I saw it was a lambda